src/app/components/fabric-post/fabric-post.component.ts

```diff
@@ -1,117 +1,157 @@
 import { Component, Inject } from '@angular/core';
 import { fabric } from 'fabric';
-import { POST_DEFAULT_BORDER, POST_DEFAULT_BORDER_THICKNESS } from 'src/app/utils/constants';
+import {
+  POST_DEFAULT_BORDER,
+  POST_DEFAULT_BORDER_THICKNESS,
+} from 'src/app/utils/constants';
+import { FabricUtils } from 'src/app/utils/FabricUtils';
 
-const AUTHOR_OFFSET = 65
-const DESC_OFFSET = 80
-const CONTENT_EXTRA_HEIGHT = 55
+const AUTHOR_OFFSET = 65;
+const DESC_OFFSET = 80;
+const CONTENT_EXTRA_HEIGHT = 55;
 
 @Component({
   selector: 'app-fabric-post',
   templateUrl: './fabric-post.component.html',
-  styleUrls: ['./fabric-post.component.scss']
+  styleUrls: ['./fabric-post.component.scss'],
 })
 export class FabricPostComponent extends fabric.Group {
+  constructor(
+    @Inject(Object) options: any,
+    ) {
+    var fabricUtils: FabricUtils = new FabricUtils();
```

Comment:
Instead of instantiating it here, you should inject it through the constructor. 

See 'canvas.component.ts' constructor for example.

Reply:
@Ammar-T I was trying to inject the fabricUtils before like this as well: 
```js
constructor(
    @Inject(Object) options: any,
    protected fabricUtils: FabricUtils,
  )
```
But this was giving and error because we we're exporting the FabricPostComponent and using it in other classes and when we initialize the FabricPostComponent class it gives the following error: `An argument for 'fabricUtils' was not provided.` I assume this is because we do not give it the fabricUtils in the constructor. Would I have to pass the fabricUtils when initializing the `FabricPostComponent`. 
I was unsure about that since I didn't see something like that in the code so that is why I did it the other way.
Maybe I'm doing it wrong. What would be the correct way of doing this? Thanks.

Reply:
I see what you mean. I think leave it how you have it for now since that works. 

We'll need to redo how we create Fabric posts anyways. It doesn't make sense to have a whole angular component for it since we're just doing all the logic within the constructor lol. 

Once we change that, we'll find a nicer way of accessing fabric utils rather than forcing it through the constructor which doesn't make a lot of sense.